src/core/prometheus.cc

```diff
@@ -89,6 +89,57 @@ static pm::Metric* add_label(pm::Metric* mt, const metrics::impl::labels_type &
     return mt;
 }
 
+static void fill_old_type_histogram(const metrics::histogram& h, ::io::prometheus::client::Histogram* mh) {
+    mh->set_sample_count(h.sample_count);
+    mh->set_sample_sum(h.sample_sum);
+    for (auto b : h.buckets) {
+        auto bc = mh->add_bucket();
+        bc->set_cumulative_count(b.count);
+        bc->set_upper_bound(b.upper_bound);
+    }
+
+}
+
+static void fill_native_type_histogram(const metrics::histogram& h, ::io::prometheus::client::Histogram* mh) {
+
+    mh->set_sample_count(h.sample_count);
+    mh->set_sample_sum(h.sample_sum);
+    size_t id = h.min_id.value();
+
+    mh->set_schema(h.schema.value());
+    double last_bucket = 0;
+    double count = 0;
+
+    size_t length = 0;
+    size_t last_bucket_id = 0;
+    ::io::prometheus::client::BucketSpan* bc = nullptr;
```

Comment:
could you add a comment explain what this `bc` for? BTW, i think `bc` deserves a better name. what `bc` stands for?

Reply:
Added documentation and renamed for clarity